src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -3523,50 +3500,25 @@ GenTree* Lowering::LowerHWIntrinsicCndSel(GenTreeHWIntrinsic* cndSelNode)
 {
     assert(cndSelNode->OperIsHWIntrinsic(NI_Sve_ConditionalSelect));
 
-    GenTree* op1 = cndSelNode->Op(1);
-    GenTree* op2 = cndSelNode->Op(2);
-    GenTree* op3 = cndSelNode->Op(3);
+    GenTree* op1         = cndSelNode->Op(1);
+    GenTree* op2         = cndSelNode->Op(2);
+    GenTree* op3         = cndSelNode->Op(3);
+    GenTree* lowerCndSel = cndSelNode;
 
     if (op1->OperIsHWIntrinsic())
     {
         GenTreeHWIntrinsic* predNode = op1->AsHWIntrinsic();
-        if (predNode->IsAllFalse())
-        {
-            // Transform:
-            //
-            // CndSel(AllFalse, op2, op3) to
-            // op3
-            op1->SetUnusedValue();
-            op2->SetUnusedValue();
-            cndSelNode->SetUnusedValue();
+        // op1 should never be Zero, because we optimized that in HIR already
+        // op1 should never be a SIMD node, because op1 takes TYP_MASK
+        assert(!predNode->IsMaskZero() && varTypeIsMask(predNode));
```

Comment:
I don't think "never" is accurate, as there may be cases (such as `minopts`) where we refuse to fold them.

I rather think it's just not something we need to be concerned about for typical compilations as we expect it to have been handled already.